src/processpriv.h

```diff
@@ -56,22 +77,246 @@ class ProcessPriv : public AbstractPrivateData {
     DLLLOCAL static void waitForTermination(int pid, ExceptionSink* xsink);
 
 private:
-    bp::child *m_process;
-    bp::opstream m_in;
-    bp::ipstream m_out;
-    bp::ipstream m_err;
+    ResolvedCallReferenceNode* optsExecutor(const char* name, const QoreHashNode* opts, ExceptionSink* xsink);
 
-    const ResolvedCallReferenceNode* optsExecutor(const char * name, const QoreHashNode *opts, ExceptionSink *xsink);
+    bp::environment optsEnv(const QoreHashNode* opts, ExceptionSink* xsink);
+    const char* optsCwd(const QoreHashNode* opts, ExceptionSink* xsink);
 
-    bp::environment optsEnv(const QoreHashNode *opts, ExceptionSink *xsink);
-    const char* optsCwd(const QoreHashNode *opts, ExceptionSink *xsink);
+    //! Process exe arguments passed through constructor.
+    void processArgs(const QoreListNode* arguments, std::vector<std::string>& out);
+    
+    //! Prepare stdin ASIO buffer for use in async_write operation.
+    void prepareStdinBuffer();
 
 #ifdef __linux__
     DLLLOCAL static QoreHashNode* getMemorySummaryInfoLinux(int pid, ExceptionSink* xsink);
 #endif
 #if defined(__APPLE__) && defined(__MACH__)
     DLLLOCAL static QoreHashNode* getMemorySummaryInfoDarwin(int pid, ExceptionSink* xsink);
 #endif
+
+    //! Thread-safe input buffer. Used for writing to child process's stdin.
+    class InputBuffer {
```

Comment:
please make the class final, or define destructor virtual. Also I would like to obey the rule of five for new classes, so define copy/move ctors/=